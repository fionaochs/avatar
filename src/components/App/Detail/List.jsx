import React from 'react';
import PropTypes from 'prop-types';
import Detail from './Detail';

const List = ({ characters }) => {
  const charactersList = characters.map(character => (
    <li key={character.id}>
      <Detail {...characters} />
    </li>
  ));
  return (
    <ul>
      {charactersList}
    </ul>
  );
};

List.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    allies: PropTypes.string.isRequired,
    affiliation: PropTypes.string.isRequired,
    enemies: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
  })).isRequired
};

export default List;
