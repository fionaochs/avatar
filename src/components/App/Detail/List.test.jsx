import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

const characters = [
  {
    id: 1,
    name: 'Ang',
    allies: 'all',
    affiliation: 'air', 
    enemies: 'fire nation',
    image: 'url'
  }
];

describe('List', () => {
  it('matches a List snapshot', () => {
    const wrapper = shallow(<List characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
