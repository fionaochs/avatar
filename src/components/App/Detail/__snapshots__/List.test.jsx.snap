// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches a List snapshot 1`] = `
<ul>
  <li
    key="1"
  >
    <Detail
      0={
        Object {
          "affiliation": "air",
          "allies": "all",
          "enemies": "fire nation",
          "id": 1,
          "image": "url",
          "name": "Ang",
        }
      }
    />
  </li>
</ul>
`;
